### ISSUE TITLE
Improving Delegators and chain Safety

### ISSUE THREAD

**Issue by stevenoruzi** [atomone-hub/genesis#74]
1. Validators acting maliciously or negligently leading to slashing predominantly affects delegators, a significant concern within the Cosmos ecosystem. Many validators merely meet the minimal self-bond requirement, a practice that can be detrimental. While delegators should ideally choose the most reliable validators, they often lack the necessary expertise to do so. Smaller delegators, in particular, tend to trust validators implicitly, without frequent verification. In such cases, it is the delegators, not the validators, who bear the brunt of any loss in value. To address this, one potential solution is a variable minimum self-bond requirement. Validators could be required to maintain a self-bonded token amount equal to the potential slashing penalty (thus directly linking their stake to their behavior) or a significant percentage, such as 50%. Additionally, validators might be restricted from withdrawing commissions if their self-bond falls below this floating minimum.
The case of the Notional validator on Osmosis exemplifies this issue. Despite its inactivity and jailed status, it still holds 3.83 million bonded tokens, underscoring the urgent need for tighter control and greater awareness among delegators.

2. The practice of creating new validators with the same name after one is jailed, as observed in some chains, should be disallowed. This would prevent confusion and potential misuse of trust.

3. To reduce risks in the chain, imposing a cap on the amount of bonded tokens held by a single validator might be effective. For example, limiting a validator to a maximum of 10% of staked tokens would prevent excessive centralization of power. Once this threshold is reached, no further delegations would be permitted, encouraging delegators to seek alternative validators, thus promoting a more equitable and secure network.

4. Finally, addressing the issue of jailed validators, they should not be allowed to undelegate their tokens. It would also be prudent to implement a comprehensive undelegation function for validators. In cases where a validator is jailed, they could issue a full undelegation request to the chain, automatically returning all delegated tokens to the respective owners' wallets. This would enhance the security and fairness of the staking process.

**Comment by jaekwon** [comment]
> In such cases, it is the delegators, not the validators, who bear the brunt of any loss in value.

and the validators could have no self-delegation at all... 

> To address this, one potential solution is a variable minimum self-bond requirement.

or we could just display the self-bond amount, and, what if somebody wants to delegate anyways even if the self-bond was small? this could be the case for any validator just getting started... startups eat ramen for dinner, speaking from experience. 

if somebody wanted to delegate to a validator that had no self-delegation at all, then shouldn't they be free to? perhaps the validator is a general partnership and the stakers have their own addresses. 

If we force self-delegation minimums, it would create a situation where people would pool tokens when they otherwise before didn't have to, or they would be forced to get a loan, again when they don't want to. The result is that it doesn't accomplish what you want, and it creates worse distortions. 

That said, how about the following? 

- [ ] Make self-staking take longer to unbond. Say twice as long.
- [ ] With the above ensure that delegators can unstake first if self-staking decreases.
- [ ] Create tooling to alert stakers about self-staking changes.

**Comment by jaekwon** [comment]
> To reduce risks in the chain, imposing a cap on the amount of bonded tokens held by a single validator might be effective. For example, limiting a validator to a maximum of 10% of staked tokens would prevent excessive centralization of power. Once this threshold is reached, no further delegations would be permitted, encouraging delegators to seek alternative validators, thus promoting a more equitable and secure network.

a 10% cap is interesting. this would be a rule that prohibits any validator that already has 10% from receiving more.
at genesis, it could be 25% each, but a new validator would not be able to achieve 1/5 because that would be 20%. so maybe as an exception, anyone can grow to become as large as the second smallest validator, because that only helps decentralization and is a simple rule that makes sense with genesis. 

We can also put a little bit of wiggle room so that there can be a bit of over-subscription, and under-subscription AND a grace period before a validator is removed from the set. Say the first 25% of over-subscription is only nominally punished, but then any amount above 25% is progressively punished for new stakers, and over-subscribers become regular subscribers when there is available room (it is a waiting queue system), and when there over-subscription, the original subscribers can be incentivized by over-subscribers to leave. And when there is over-subscription, there is more slashed than the nominal maximum amount (they are all still at stake). And over-subscribers otherwise have full voting rights. 

I think we could maybe do better... why not a 1% cap? Then there would be many top level validators. The cap should be determined ultimately (when t is in the far future) by how many equal full sized validators we want in the world. There are 195 countries in the world, so a 0.33% cap is compatible with every country having at least one full validator.

**Comment by ABCharlieEth** [comment]
Interesting idea to achieve decentralization, maybe you are aware that Sifchain implemented this in 2022, the implemented with 6% and the aim was to reduce it to 3%. But, issue they were facing at the moment was the resistance from the top validators  among other issues. I believe we should start from lower than 10% as it will be hard to reduce it much in the later stage. 

Ideally, for a perfect scenario all val set should be global and equally talented with equal VP, but due to many reasons we can not achieve this but we should try to get close to the target as much as possible. On code level if we could implement the max-vp-cap we should, and cap should be lower than 10% from the start with the intention the go lower in the future.

**Comment by jaekwon** [comment]
Yes and if we just have the right dashboard it should be good to incentivize world-wide decentralization and participation. The stakers can choose for themselves, or else risk a split doing it better.

**Comment by hsibai** [comment]
I propose a variable rewards system affecting the rewards a validator receives based of preset rules such as:

1. A validator whose self bonded to delegated ratio is below a preset min parameter received less rewards. The clawed/taxed amount goes to comm pool. 
2. A validator whose total bonding grows to more than 1% (for example) of supply also receives diminishing rewards on the excess amount above 1%. Aim is to discourage centralization.
3. The above will encourage them to remain in "The Goldilocks Zone" to receive 100% of the rewards.

**Comment by stevenoruzi** [comment]
> or we could just display the self-bond amount, and, what if somebody wants to delegate anyways even if the self-bond was small? this could be the case for any validator just getting started... startups eat ramen for dinner, speaking from experience.
``
you are right, it might seem unfair to the small fish. We could consider offering support to them in the Genesis phase, perhaps through an airdrop to validators involved in the testing phase. Additionally, we could introduce a new risk factor on UI that is a result of multiple parameters such as missed blocks, self-bond, participating governance (, etc.) to make it more practical for the users.

> That said, how about the following?
> 
> * [ ]  Make self-staking take longer to unbond. Say twice as long.
> * [ ]  With the above ensure that delegators can unstake first if self-staking decreases.
> * [ ]  Create tooling to alert stakers about self-staking changes.

I think none of them would work. also as you said stringent minimum requirements on self-bond won't work as we want. I would like to find a better validator guarantor. or we should leave this part and try to make it simpler for the user from UI to decrease their risk as mentioned above.

**Comment by jaekwon** [comment]
> what if somebody wants to delegate anyways even if the self-bond was small

@stevenoruzi @hsibai on second thought, I think the workaround to the above is simply to lend a small portion of the ATOMs (you would otherwise delegate) to the validator you want to support. This could guarantee that the validator has skin in the game, but this depends on how the loan is constructed.

Even if the loan were denominated in ATONEs AS WELL AS all the rewards that could be earned by staking them, then the validator in effect does not have any positive skin in the game; but there is negative (punitive) skin (stake) because slashing would incur a loss against the debt. To remove the validator's skin in the game with a loan, you would have to go through the extra step of forgiving slashes when constructing the loan. But is the negative skin enough? Not enough if the validator (or a set of them) can collude to censor txs or make illegal transitions in their favor with an exit attack. So in that sense it is inferior because the attack makes the loan moot (it will never get repaid anyways). In the absence of that possibility it is safe, but without discernment of new startup validators, it doesn't ensure anything against attacks. Other types of loans like private loans against the dollar would help if the loan terms allowed for recompense in case of fraud or recklessness, as enforced by jurisdictions of your choice.

In short, I'm not opposed to the self-stake minimum. So validators are putting stake where their own signer is. But what range of % is reasonable? from first-principles, or from empirical comparables (from the old world)?

**Comment by jaekwon** [comment]
minimum self-bond ratio will create a marketplace of loans where stakers loan some amount for the validator to "self-delegate", for some slightly less than favorable terms as opposed to the validator simply staking themselves. If the terms were significantly less favorable for the validator, somebody will offer a competing service to validators. This is easy to do with ICA, and easy enough to do with the existing legal system. These loans would not be traceable in general, and we would lose any insight from self-delegation ratios. So I think this is practically self-defeating.

Validators *do* have something at stake; they have the future positive revenue/inflation at stake. Arguably this is fleeting as the delegators may undelegate at any time, but if the unbonding period were optionally allowed to be significantly longer, then this is arguably a stronger guarantee and it is also open to anyone. But I don't think I would want to require a minimum % to be long-bonded either because of the distortions that it might introduce. We could disable long-bonding from ICA accounts to discourage distortion markets (a problem which doesn't exist with "self-bonding"). And only then, might I say, that I think sidechannel (off chain) distortion markets are less likely to appear with a minimum long-bond % as opposed to a minimum self-bond %, and so less self-defeating, but I'm not sure.

**Comment by COverS8** [comment]
In Cosmos Hub, many delegators usually delegates to the lowest commission validators to maximise their rewards. What if AtomOne Hub validators were set the same commission rate that could not be changed without a proposal and the commission change proposal could not apply to just one particular validator but to the whole set, plus a scoring system based on reputation that would give an indication of a validator's reliability. The scores would not rank the validators e.g. 1-50 to avoid delegators to delegating just for the Top validators but e.g. in two separate baskets. Basket 1 for validators who performs their duty (uptime/creating blocks) and basket 2 for validators who don't, misses the blocks/slashed/jailed. Another duty of the validator is to participate in the governance which should also be linked to the scoring. By eliminating the advance with the commission, the decision of the delegators would be based on the reliability of the validator which should be visible to delegators when making a delegation decision, either directly in the wallet if possible or on the dashboard. The delegators can then decide between the basket 1 validators where to base delegation by this scoring system and also by doing own research by visiting on validators websites and social medias to see how dedicated and engaged the validators are with their contributions to the Cosmos Network and AtomOne. Ending up in Basket 2 would incentivised the validators to perform better and the basket may even be empty if everyone performs their duty.

**Comment by ABCharlieEth** [comment]
> Yes and if we just have the right dashboard it should be good to incentivize world-wide decentralization and participation. The stakers can choose for themselves, or else risk a split doing it better.

Can you explain what you meant when you said the "the right dashboard"?
